frontend/src/utils/helpers.js

```diff
@@ -0,0 +1,66 @@
+const reportAction = {
+  action: 'report',
+  label: 'Report'
+}
+
+const ACTIONS_MAP = {
+  question: {
+    canDelete: {
+      action: 'deleteQuestion',
+      label: 'Delete'
+    },
+    canEdit: {
+      action: 'editQuestion',
+      label: 'Edit'
+    }
+  },
+  review: {
+    canDelete: {
+      action: 'deleteReview',
+      label: 'Delete'
+    },
+    canEdit: {
+      action: 'editReview',
+      label: 'Edit'
+    }
+  },
+  comment: {
+    canDelete: {
+      action: 'deleteComment',
+      label: 'Delete'
+    },
+    canEdit: {
+      action: 'editComment',
+      label: 'Edit'
+    }
+  }
+}
+
+/**
+ * Given a parent component type ['review', 'question']
+ * Map callback handlers and labels for menu to set types.
+ */
+export function menuInteractionsMapper({
+  type,
+  thisArg
+}) {
+  if (!thisArg.meta) return
```

Comment:
i'm not a fan of returning undefined here ... i see why you did it but having a function return two different types feels icky to me

Reply:
Yeppp